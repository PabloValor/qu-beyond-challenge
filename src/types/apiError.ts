export interface ApiError { 
    message: string
    reason?: string
    level?: number
    trace?: string
}