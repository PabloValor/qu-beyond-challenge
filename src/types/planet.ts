import Film from "./film"
import People  from "./people"

export interface Planet {
    id?: string
    name: string
    diameter: string
    rotation_period: string
    orbital_period: string
    gravity: string
    population: string
    climate: string
    terrain: string
    surface_water: string
    residents: People[]
    films: Film[]
    url: string
    created: string
    edited: string
}

export interface PlanetResponse { 
    count: number
    next: string
    previous: string
    results: Planet[]
}