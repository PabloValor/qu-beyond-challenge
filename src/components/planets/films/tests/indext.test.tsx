import { MemoryRouter } from "react-router-dom";
import Film from "..";
import { default as TFilm} from '../../../../types/film'
import { ReactTestRenderer, ReactTestRendererJSON, create } from "react-test-renderer";
import { render } from "@testing-library/react";

const filmMock: TFilm = {
    "title": "A New Hope", 
    "episode_id": 4, 
    "opening_crawl": "test", 
    "director": "George Lucas", 
    "producer": "Gary Kurtz, Rick McCallum", 
    "release_date": "1977-05-25", 
    "characters": [
    ], 
    "planets": [
    ], 
    "starships": [
    ], 
    "vehicles": [
    ], 
    "species": [
    ], 
    "created": "2014-12-10T14:23:31.880000Z", 
    "edited": "2014-12-20T19:49:45.256000Z", 
    "url": "https://swapi.dev/api/films/1/"
}

function toJson(component: ReactTestRenderer) {
const result = component.toJSON()
expect(result).toBeDefined()
return result as ReactTestRendererJSON
}

describe('<Film />', () => {

    test('should match snapshot', () => { 
        const component = create(
        <MemoryRouter>
            <Film films={[filmMock]}/>
        </MemoryRouter>
        )
        let json = toJson(component)
        expect(json).toMatchSnapshot() 
    })

    test('Planet mounts properly', () => {
        const wrapper = render(<MemoryRouter> <Film films={[filmMock]}/></MemoryRouter> )
        expect(wrapper).toBeTruthy()

        const span = wrapper.container.querySelector('span')
        expect(span?.textContent).toBe(filmMock.title)
    })
});