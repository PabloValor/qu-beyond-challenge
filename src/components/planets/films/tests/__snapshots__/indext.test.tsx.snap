// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Film /> > should match snapshot 1`] = `
<div
  className="bg-slate-800 text-slate-100 p-1 rounded-lg shadow-md w-full mb-3 md:w-1/4 md:mb-0"
>
  <div
    className="m-2 uppercase text-center"
  >
    <svg
      className="inline-block mr-1 text-2xl"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M14 5v14h-4V5h4m6-2h-2v2h-2V3H8v2H6V3H4v18h2v-2h2v2h8v-2h2v2h2V3zm-4 6V7h2v2h-2zM6 9V7h2v2H6zm10 4v-2h2v2h-2zM6 13v-2h2v2H6zm10 4v-2h2v2h-2zM6 17v-2h2v2H6z"
      />
    </svg>
    <span
      className="font-bold"
    >
      A New Hope
    </span>
  </div>
  <div
    className="m-2"
  >
    Director: 
    <span
      className="font-bold"
    >
      George Lucas
    </span>
  </div>
  <div
    className="m-2"
  >
    Producer: 
    <span
      className="font-bold"
    >
      Gary Kurtz, Rick McCallum
    </span>
  </div>
</div>
`;
